src/main/java/core/basesyntax/dao/impl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,218 @@
+
+            return optional;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.car_id, cars.model, cars.manufacturer_id,"
+                + " manufacturers.manufacturer_country, manufacturers.manufacturer_name "
+                + "FROM cars "
+                + "Inner join manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.manufacturer_id "
+                + "Where cars.deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            return getCars(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars ", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE car_id = ? AND deleted = false ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setLong(2, car.getManufacturer().getId());
+            preparedStatement.setLong(3, car.getId());
+            preparedStatement.executeUpdate();
+            deleteDriversForCar(car.getId());
+            addDriversForCar(car);
+            int updateRows = preparedStatement.executeUpdate();
+            if (updateRows > 0) {
+                return car;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update this car " + car, e);
+        }
+        throw new DataProcessingException("No car with id " + car.getId());
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET deleted = true WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            deleteDriversForCar(id);
+            int updateRows = preparedStatement.executeUpdate();
+            return updateRows > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = " SELECT c.car_id as car_id,"
+                + " c.model as model,"
+                + " c.manufacturer_id as manufacturer_id,"
+                + " m.manufacturer_country as manufacturer_country, "
+                + " m.manufacturer_name as manufacturer_name "
```

Comment:
we should use `as something` when columns have same names in different tables

`"SELECT c.id AS car_id, c.model, m.id AS manufacturer_id...`

here it is redundant